apps/server/src/modules/tool/service/external-tool.service.ts

```diff
@@ -128,4 +129,11 @@ export class ExternalToolService {
 			return true;
 		});
 	}
+
+	isRegexCommentMandatoryAndFilled(customParameter: CustomParameterDO): boolean {
```

Comment:
I think the arc group decided to have logic in domain object. Because of that and personally preferences move this method into the domain object, so the domain object itself can be asked. Furthermore this looks like a validation.. in this case use our validation service and there call the method on the object :)

Reply:
done